From: Pino Toscano <pino@kde.org>
Subject: Fix platform file for GNU/Hurd
 Currently (both in cmake 2.6.x and cmake 2.8 as of its latest rc5), the
 GNU.cmake platform file used for GNU/Hurd systems is not complete.  In
 particular, it needs the rpath-link flag definition, needed for properly
 compiling in all the cmake setups; in particular, its miss causes the failure
 of the ExportImport unit test, which instead works correctly with the proposed
 patch.  Also, the patch provides some more definitions, a bit close to the
 Linux oones (given the base toolchain is quite close anyway, GNU-based).
Bug: http://public.kitware.com/Bug/view.php?id=9873
Forwarded: http://public.kitware.com/Bug/view.php?id=9873
Origin: other, http://public.kitware.com/Bug/file_download.php?file_id=2644&type=bug

--- a/Modules/Platform/GNU.cmake
+++ b/Modules/Platform/GNU.cmake
@@ -4,5 +4,10 @@
 SET(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-shared")
 SET(CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG "-Wl,-rpath,")
 SET(CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG_SEP ":")
+SET(CMAKE_SHARED_LIBRARY_RPATH_LINK_C_FLAG "-Wl,-rpath-link,")
 SET(CMAKE_SHARED_LIBRARY_SONAME_C_FLAG "-Wl,-soname,")
+SET(CMAKE_SHARED_LIBRARY_SONAME_Fortran_FLAG "-Wl,-soname,")
+SET(CMAKE_EXE_EXPORTS_C_FLAG "-Wl,--export-dynamic")
+SET(CMAKE_EXE_EXPORTS_CXX_FLAG "-Wl,--export-dynamic")
+
 INCLUDE(Platform/UnixPaths)
